RaisingTheBar.asciidoc

```diff
@@ -1,34 +1,70 @@
 
 === Static code analysis
 
-Static code analysis is where the code of a solution is analysed without it having to be build and executed. It ensures that the code adheres to industry and team standards. Performing static code analysis can:
+Static code analysis is where the code of a solution is analyzed without it having to be built and executed. It ensures that the code adheres to industry and team standards. Performing static code analysis can:
 
 *	Improve the ability for developers to understand each other’s code
 *	Reduce the time needed for a peer review
 *	Catch issues that may not otherwise be caught
 
-=== Automated UI testing
+===	A/B testing
+
+A/B testing allows you to set up two deployment slots (A and B) in production where one of those slots is used to deploy a new release to. So you have a hot slot and a staging slot. Using traffic routing, you can then send a percentage of the traffic to the new slot. That way, without affecting all users, you can have a period of testing in production where any new introduced issues should not affect all users.
+
+A/B testing also allows high performance sites to always be available with zero warm up time needed. You deploy to the staging slot, start the slot running, and once running, swap this slot to be the production slot.
+
+You can also look at implementing an intelligent solution that will analyse the logs and dynamically amend the routing rules to pass users to the new code on confirmation that there are no detrimental issues affecting performance or correct behaviour. Or, alternatively, fall back to the old code if an issue is detected.
+
+=== Automated UI and API testing
+
+To speed up the repeatable nature of testing to help improve the cadence of release, where possible we should leverage the use of automated test tools such as Playwright or Selenium Web Driver.
+
+=== CI/CD Pipelines
+
+Automation is crucial to maintaining a high level of efficiency and quality in fast-moving projects. Incorporating automated tests into CI/CD pipelines enables early bug detection and helps validate every code change. These tests should span from unit tests to integration and regression tests, ensuring that new updates don't break existing functionality.
+
+=== Exploratory and Context-Driven Testing
 
-To speed up the repeatable nature of testing to help improve the cadence of release, where possible we should leverage the use of automated test tools such as Selenium Web Driver .
+Beyond automation, exploratory testing is invaluable for identifying edge cases and unexpected behavior. Testers use their knowledge of the system to explore and test functionality in unscripted ways, often revealing bugs that standard scripted tests may miss. Coupled with context-driven testing, this approach allows testers to adapt based on the specific requirements and nuances of each project, making the testing process more flexible and responsive.
 
 === Security assessment & compliance
 
```

Comment:
Can you also amend the text "Consider security during the security and compliance using automated tools such as ..." (as "security during the security" sounds odd (I know that wasn't you, but while we are addressing this section.

Reply:
Fixed